CPP-187 - Enhance Integration Tests for Cassandra v1.2.x

mikefero opened the pull request:
Integration tests that could run against C* v1.2.x have been fixed along with the skipping of batch and serial_consistency suites.

mpenick commented:
@mikefero I would prefer is we only use prepared statements if the version <= 1.2, otherwise we should use a bound statements.

mikefero commented:
@mpenick Addressed comments and brought inline with the 2.1.0-beta by updating to master and cherry picking commits for v3 protocol in CPP-260/PR-170. Force committed to remove previous version this PR to ensure no merge conflict resolution would be needed (manual merged previous changes and addressed API changes where needed)

mpenick commented:
Merged in 20d457c295f208a15f9ddba425f7e2ea00d74329
